#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_0179-transient-hack.dpatch by Lubos Lunak <l.lunak@kde.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: qt-bugs issue : 166027
## DP: This patch is just a hackish workaround that makes setting of WM_TRANSIENT_FOR
## DP: work with some window types (needed basically by the 0178 patch).
## DP: It is not a proper solution, waiting for TT to come up with something.

@DPATCH@
diff -urNad qt4-x11-4.3.0~/src/gui/kernel/qwidget_x11.cpp qt4-x11-4.3.0/src/gui/kernel/qwidget_x11.cpp
--- qt4-x11-4.3.0~/src/gui/kernel/qwidget_x11.cpp	2007-05-25 15:24:19.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/kernel/qwidget_x11.cpp	2007-06-26 20:01:42.000000000 +0200
@@ -1569,7 +1569,7 @@
         do_size_hints(q, extra);
 
         // udpate WM_TRANSIENT_FOR
-        if (isTransient(q)) {
+        if (isTransient(q) && !( q->windowFlags() & Qt::X11BypassWindowManagerHint)) {
             QWidget *p = q->parentWidget();
             if (p)
                 p = p->window();
